core/src/lib.rs

```diff
@@ -50,6 +50,22 @@ pub use edn::{
 /// use i64 rather than manually truncating u64 to u63 and casting to i64 throughout the codebase.
 pub type Entid = i64;
 
+/// An entid that's either already in the store, or newly allocated to a tempid.
```

Comment:
I'm really struggling to understand a lot of these refactoring commits.  On the one hand, they're "clearly correct", since they could be automated.  On the other hand... why are you doing this?  What advantage do you get by wrapping an `Entid` with a marker bit about its allocation status?  There's a `PartitionMap` that captures that data...

Reply:
`KnownEntid` is a newtype, so it's eliminated by the compiler; it's not actually wrapped. What it does is make it difficult to accidentally pass a raw numeric entid, which might be out of range, to functions that will only work correctly for an in-range value. That's helpful for making sure that an entid actually came from resolving an ident against a real store, rather than using some numeric constant that might be wrong. You'll see this in `entity_builder.rs` — the `a` place must be a `KnownEntid`, so you can't just use `DB_SOME_ATTR` or `123` and have it work.